cpp/src/groupby/sort/functors.hpp

```diff
@@ -87,7 +87,6 @@ struct store_result_functor {
    */
   column_view get_sorted_values()
   {
-    if (is_presorted()) { return values; }
```

Comment:
I'm trying to grok the implication of removing this line. (Not that I disagree with the change.)

In cases (like in Spark) where the grouping keys and values are already sorted, there will be another sorting step, possibly changing the order of the values?
(I worry for cases where there are repeated values.)

Reply:
The results will again gathered in original grouped values order and returned. So, results order match. 
I am concerned with another `sort` operation even though we don't really need it when we know that values are also sorted. 
In cudf python, there is no guarantee that the values are sorted too. or even keys. (defaults to sorted::NO)

We don't have a flag for `values_are_sorted` now. If values are always sorted when keys are sorted in spark, then we can simply replace `_keys_are_sorted` with a flag that means `_keys_values_are_sorted` and replace their usages too.

Reply:
> then we can simply replace _keys_are_sorted with a flag that means _keys_values_are_sorted and replace their usages too.

Ah, this must have been what @devavret was mentioning yesterday: To use a flag to avoid re-sorting the data, in cases like Spark's. Would this be viable to add?